src/SIS_dyn_cgrid.F90

```diff
@@ -1738,17 +1781,192 @@ subroutine basal_stress_coeff_C(G, mi, ci, sea_lev, CS)
       else
         CS%Tb_v(i,J) = 0.0
       endif
-!     if (G%idg_offset + i == 471 .and. G%jdg_offset + J == 671) then
-!     if (G%idg_offset + i == 25 .and. G%jdg_offset + J == 573) then
-!       print *, 'inside basal_stress', hw_v, ci_v, h_v, hc_v
-!     endif
     enddo
   enddo
 !          call uvchksum("Tb_[uv] before SIS_C_dynamics", CS%Tb_u, CS%Tb_v, G, &
 !                         halos=1, scale=US%RZ_T_to_kg_m2s*US%L_T_to_m_s)
 
 end subroutine basal_stress_coeff_C
 
+!> Computes basal stress Tbu coefficients (landfast ice)
+!!
+!! Computes seabed (basal) stress factor Tbu (landfast ice) based on
+!! probability of contact between the ITD and the seabed. The water depth
+!! could take into account variations of the SSH. In the simplest
+!! formulation, hwater is simply the value of the bathymetry. To calculate
+!! the probability of contact, it is assumed that the bathymetry follows
+!! a normal distribution with sigma_b = 2.5d0. An improvement would
+!! be to provide the distribution based on high resolution data.
+!!
+!! Dupont, F. Dumont, D., Lemieux, J.F., Dumas-Lefebvre, E., Caya, A.
+!! in prep.
+!!
+!! authors: D. Dumont, J.F. Lemieux, E. Dumas-Lefebvre, F. Dupont
+!!
+subroutine basal_stress_coeff_itd(G, IG, IST, sea_lev, CS)
+
+  type(SIS_hor_grid_type),            intent(in)  :: G     !< The horizontal grid type
+  type(ice_grid_type),                intent(in)  :: IG    !< The sea-ice specific grid type
+  type(ice_state_type),               intent(in)  :: IST   !< A type describing the state of the sea ice
+  real, dimension(SZI_(G),SZJ_(G)),   intent(in)  :: sea_lev !< Sea level [Z ~> m]
+  type(SIS_C_dyn_CS),                 pointer     :: CS    !< The control structure for this module
+
+  integer, parameter :: &
+           ncat_b = 100, &  ! number of bathymetry categories
```

Comment:
Given that there are some operations that occur ncat_b x ncat_i times for each grid cell, it seems dangerous to specify these as (large) fixed parameters.  It is hard-coded that this routine will do 10,000 operations at every grid point that is both shallow and has enough ice.  It seems to me that these sizes should be model variables that can be determined and set at run-time.